### PR TITLE
Runtime 9.1 release notes

### PR DESCRIPTION
**Pull request by jwitz**

### CODE REVIEW

File: astro/runtime-release-notes.md [hunk 1]
```diff
@@ -24,6 +24,25 @@ Astro Runtime is a Docker image built and published by Astronomer that extends t
...
+## Astro Runtime 9.1.0
+
+- Release date: September 7, 2023
+- Airflow version: 2.7.1
+
+### Airflow 2.7.1
+
+Astro Runtime 9.1.0 includes same-day support for Apache Airflow 2.7.1. Airflow 2.7.1 contains a number of bug fixes including:
+
+[Key bug fixes here]
```
Comment (jwitz) [inline astro/runtime-release-notes.md:36]: Need some key commits here if possible. If it's all really boring bug fixes, I'll say that (in a nicer way tho)

File: astro/runtime-release-notes.md [hunk 2]
```diff
@@ -24,6 +24,25 @@ Astro Runtime is a Docker image built and published by Astronomer that extends t
...
+
+To learn more, see the [Apache Airflow 2.7.1 release notes](https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html#airflow-2-7-1-2023-09-07).
+
+### Additional improvements
+
+- Upgraded `astronomer-providers-logging` to 1.2.1.
```
Comment (jwitz) [inline astro/runtime-release-notes.md:42]: Anything we can say about what's changed here?

Comment (ephraimbuddy) [reply astro/runtime-release-notes.md:42]: Datadog log forwarding support is now also available in azure and GCP if configured. Can be configured with `ASTRO_DATADOG_TASK_LOGS_ENABLED`.

I think this is only available for hosted. cc @jedcunningham

File: astro/runtime-release-notes.md (hunk 1 above)
Comment (ephraimbuddy) [reply astro/runtime-release-notes.md:36]: https://github.com/apache/airflow/releases/tag/2.7.1

Comment (ephraimbuddy) [reply astro/runtime-release-notes.md:36]: There are some significant changes

File: astro/runtime-release-notes.md (hunk 2 above)
Comment (jedcunningham) [reply astro/runtime-release-notes.md:42]: Hmm, not sure if it is hosted only or not actually. @melugoyal?

File: astro/runtime-release-notes.md (hunk 1 above)
Comment (jedcunningham) [reply astro/runtime-release-notes.md:36]: Generally I'd say https://github.com/apache/airflow/pull/33632 is one that would be worth calling out, but it really doesn't apply to Astro anyways.

If you need a few that would be worth mentioning:
https://github.com/apache/airflow/pull/33672
https://github.com/apache/airflow/pull/33732

I'm not sure either of the changes that got a significant section are worth calling out in our release notes (especially the conf one).

File: astro/runtime-release-notes.md (hunk 2 above)
Comment (jwitz) [inline astro/runtime-release-notes.md:42]:
```suggestion
- You can now forward logs to Datadog from Azure and GCP clusters.
```

File: astro/runtime-release-notes.md (hunk 1 above)
Comment (jwitz) [inline astro/runtime-release-notes.md:36]:
```suggestion
- Treat dag-defined access_control as authoritative if defined ([#33632](https://github.com/apache/airflow/pull/33632))
- Add limit 1 if required first value from query result ([#33632](https://github.com/apache/airflow/pull/33632))
- Fix MappedTaskGroup tasks not respecting upstream dependency ([#33732](https://github.com/apache/airflow/pull/33732))
```

File: astro/runtime-release-notes.md [hunk 3]
```diff
@@ -24,6 +24,27 @@ Astro Runtime is a Docker image built and published by Astronomer that extends t
...
+
+- Treat dag-defined access_control as authoritative if defined ([#33632](https://github.com/apache/airflow/pull/33632))
+- Add limit 1 if required first value from query result ([#33632](https://github.com/apache/airflow/pull/33632))
+- Fix MappedTaskGroup tasks not respecting upstream dependency ([#33732](https://github.com/apache/airflow/pull/33732))
+
+To learn more, see the [Apache Airflow 2.7.1 release notes](https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html#airflow-2-7-1-2023-09-07).
+
+### Additional improvements
+
+- You can now forward logs to Datadog from Azure and GCP clusters.
```
Comment (jwitz) [inline astro/runtime-release-notes.md:44]:
```suggestion
- You can now forward task logs to Datadog from Azure and GCP clusters.
```

**Pull request merged**